source/server/connection_handler_impl.h

```diff
@@ -130,6 +137,36 @@ class ConnectionHandlerImpl : public Network::ConnectionHandler, NonCopyable {
     Stats::TimespanPtr conn_length_;
   };
 
+  /**
+   * Wrapper for an active accept socket owned by this handler.
+   */
+  struct ActiveSocket : public Network::ListenerFilterManager,
```

Comment:
nit: this still have to be a class, per Google Style Guide (no deviation in envoy style for this):

`Use a struct only for passive objects that carry data; everything else is a class.`

Reply:
I tried this but this happened even when I made all members public:
```
source/server/connection_handler_impl.cc: In destructor 'virtual Envoy::Server::ConnectionHandlerImpl::ActiveListener::~ActiveListener()':
source/server/connection_handler_impl.cc:72:72: error: 'std::unique_ptr<_Tp> Envoy::LinkedObject<T>::removeFromList(Envoy::LinkedObject<T>::ListType&) [with T = Envoy::Server::ConnectionHandlerImpl::ActiveSocket; Envoy::LinkedObject<T>::ListType = std::__cxx11::list<std::unique_ptr<Envoy::Server::ConnectionHandlerImpl::ActiveSocket> >]' is inaccessible within this context
     ActiveSocketPtr removed = sockets_.front()->removeFromList(sockets_);
```

Maybe there is a reason other structs like this also exist in this file?